src/utils/rm.js

```diff
@@ -0,0 +1,20 @@
+import { join } from './join'
+
+export async function rm(fs, filepath, opts = {}) {
+  const entries = await fs.readdir(filepath)
+  if (entries == null) {
+    await fs.rm(filepath)
```

Comment:
I would use here a utility that will check if filepath is directory or not like the other commented line (and do fs.rmdir or fs.unlink), and use it here. So it's clear what is calling what.

Reply:
Another idea is to use hidden `_rm` so it's clear that it's not a recursive call, because it looks weird that you in one place call `rm()` and in other `fs.rm()`.

Reply:
It may look confusing because the functions are similarly named, but they are distinctively not the same thing. `fs.rm` is effectively `unlink` with a graceful exit if the file is already gone.

Reply:
Then maybe just add a comment that `fs.rm` is basically `unlink`.

Reply:
Sure thing. I'd also be open to renaming the rm helper function since we're only ever using it to remove recursively. I think rmRecursive (or rmRecursively) makes sense.

I'll update accordingly.